torch/_dynamo/output_graph.py

```diff
@@ -1266,7 +1289,26 @@ def remove_node(self, node):
     # for SymInts that may occur in the tensor argument.
     # Remove this if https://github.com/pytorch/pytorch/issues/99007 gets
     # fixed.
-    def create_graph_input(self, name, type_expr=None, before=False):
+    def create_graph_input(self, name, type_expr=None, before=False, source=None):
+        if source is None:
+            assert (
+                self.parent is not None
+            ), "you are required to provide a source for inputs on the root tracer"
+
+        # In eager, we are generally OK with adding graph inputs whenever we
+        # want, because we take care of writing the bytecode that knows how
+        # to source all the inputs.
+        #
+        # In export, this is bad, because you want a self-contained export
+        # object which only depends on the inputs you explicitly passed to it.
+        # So we are a bit more strict about what sources can become inputs
+        # in export
+        if self.export_root:
+            if not is_from_local_source(source, allow_cell_or_freevar=False):
```

Comment:
nit: `is_from_local_source(source, allow_cell_or_freevar=(not self.export_root))`

Reply:
no, I don't want to do this test at all when not self.export_root